File: src/main/java/net/igneo/icv/entity/helmet/blackHole/BlackHoleModel.java
package net.igneo.icv.entity.helmet.blackHole;

import net.igneo.icv.ICV;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.model.GeoModel;

public class BlackHoleModel extends GeoModel<BlackHoleEntity> {
    private final ResourceLocation model = new ResourceLocation(ICV.MOD_ID, "geo/blackhole.geo.json");
    private final ResourceLocation texture = new ResourceLocation(ICV.MOD_ID, "textures/entity/blackhole.png");
    private final ResourceLocation animations = new ResourceLocation(ICV.MOD_ID, "animations/blackhole.animation.json");
    
    
    @Override
    public ResourceLocation getModelResource(BlackHoleEntity blackHoleEntity) {
        return this.model;
    }
    
    @Override
    public ResourceLocation getTextureResource(BlackHoleEntity blackHoleEntity) {
        return this.texture;
    }
    
    @Override
    public ResourceLocation getAnimationResource(BlackHoleEntity blackHoleEntity) {
        return this.animations;
    }
}

File: src/main/java/net/igneo/icv/entity/helmet/blackHole/BlackHoleModel.kt
package net.igneo.icv.entity.helmet.blackHole

import net.igneo.icv.ICV
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.model.GeoModel

class BlackHoleModel : GeoModel<BlackHoleEntity?>() {
    private val model = ResourceLocation(ICV.MOD_ID, "geo/blackhole.geo.json")
    private val texture = ResourceLocation(ICV.MOD_ID, "textures/entity/blackhole.png")
    private val animations = ResourceLocation(ICV.MOD_ID, "animations/blackhole.animation.json")


    override fun getModelResource(blackHoleEntity: BlackHoleEntity?): ResourceLocation {
        return this.model
    }

    override fun getTextureResource(blackHoleEntity: BlackHoleEntity?): ResourceLocation {
        return this.texture
    }

    override fun getAnimationResource(blackHoleEntity: BlackHoleEntity?): ResourceLocation {
        return this.animations
    }
}
